### DISCUSSION TITLE
Determine if types behind trait objects affect stability

### DISCUSSION THREAD

**Discussion by seanmonstar** [rust-lang-nursery/api-guidelines#96] [API Guidelines]
The specific case I have in mind is `&Error + Send + Sync + 'static`. A user can downcast the trait object, and if it were an error from an unstable crate, does that violate the stability guarantee. The stable crate changing it's internal dependency wouldn't cause end users' program to stop compiling, but instead to just silently stop catching the expected error.

For an example: since a `reqwest:Error` could contain a `hyper::Error`, which isn't stable yet, should the stable crate expose a trait object containing the unstable type?

Specifically for `Error`, if it is determined that this violates stability, should those unstable errors not be returned at all?

**Comment by KodrAus** [comment]
See also: https://github.com/rust-lang/api-guidelines/discussions/162

**Comment by nox** [comment]
I'm of the opinion that it is indeed part of the public API, as things would not behave the same. Semver isn't supposed to be about whether the program still builds.
